File: app/src/main/java/com/katkov/lolachievements/domain/usecase/LoginUseCase.java
package com.katkov.lolachievements.domain.usecase;

import com.katkov.lolachievements.domain.model.EntryInfoModel;
import com.katkov.lolachievements.domain.service.LoginService;

import javax.inject.Inject;
import javax.inject.Singleton;

@Singleton
public class LoginUseCase {

    private final LoginService loginService;

    @Inject
    public LoginUseCase(LoginService loginService) {
        this.loginService = loginService;
    }

    public void saveEntryInfo(EntryInfoModel entryInfoModel) {
        loginService.saveEntryInfo(entryInfoModel);
    }

    public void removeEntryInfo() {
        loginService.removeEntryInfo();
    }
}

File: app/src/main/java/com/katkov/lolachievements/domain/usecase/LoginUseCase.kt
package com.katkov.lolachievements.domain.usecase

import com.katkov.lolachievements.data.local.service.LoginService
import com.katkov.lolachievements.domain.model.LoginModel
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class LoginUseCase
@Inject
constructor(private val loginService: LoginService) {

    fun saveLoginModel(loginModel: LoginModel) {
        loginService.saveLoginModel(loginModel)
    }

    fun removeLoginModel() {
        loginService.removeLoginModel()
    }
}
